graphql/index.ts

```diff
@@ -1,10 +1,34 @@
+const mockCachedBooks = [
+  {id: '1', name: 'default book', authors: ['author1', 'author2']}
+]
+
 export const resolvers = {
   Query: {
-    sampleStaticResolver: (_, body) => {
-      console.log(body)
-      return {
-        result: 'static with param: ' + body.param
+    cachedBooks: (_, __) => {
+      return mockCachedBooks
+    },
+    cachedBook: (_, {id}) => {
+      return mockCachedBooks.find((book) => book.id === id)
+    },
+  },
+  Mutation: {
+    deleteCachedBook: (_, {id}) => {
+      const index = mockCachedBooks.findIndex((book) => book.id === id)
+
+      if (-1 < index && index < mockCachedBooks.length) {
```

Comment:
You should only need to check if `index !== -1` here. When that's not the case, `findIndex` will always return a valid index, so no need to revalidate.

Reply:
Thanks, didn't know about it